sessions/nowcasting.qmd

```diff
@@ -31,6 +32,555 @@ The code in this session can be run as an interactive notebook using RStudio, or
+
+Why then, you might ask, not just plot the data by date of reporting which correctly showed the data to be still increasing and should, by definition, not be subject to future changes?
+This can someimtes be a sensible way to visualise the data.
+However, reporting might itself be subject to biases such as breaks during the weekend, holidays etc.
+At the same time, when it comes to capacity or intervention planning we may need to know how many people e.g. become sick on any given day and will thus present to the healthcare system rather than how many will be reported.
+Estimating the "true curve" (i.e. what we expect to see once the data are complete at a future date) of the time series of _epidemiologically relevant events_ from a potentially truncated epidemiological curve and information about the delays is what is usually called "nowcasting".
+
+# Nowcasting with a known delay
+
+## The simplest possible nowcasting model
+
+Here we assume that the delay distribution is known and that we can use it to nowcast the most recent data. In practice, the delay distribution is often not known and needs to be estimated from the data. We could do this using methods from [the session on biases in delay distributions](sessions/biases-in-delay-distributions.qmd).
+
+Previous session we used delay distributions convolved with the infection times to estimate the time series of symptom onsets. [session on convolutions](using-delay-distributions-to-model-the-data-generating-process-of-an-epidemic#estimating-a-time-series-of-infections). A simple way to nowcast is to use the same approach but using the cumulative distribution function of the delay distribution rather than the probability density function and only apply it to the most recent data as this is the only data that can be subject to change (due to delays in reporting). We will build intuition for this as usual using simulation. First we define the proportion reported using a delay distribution:
+
+```{r}
+proportion_reported <- plnorm(1:15, 1, 0.5)
+plot(proportion_reported)
+```
+
+This is similar to the `rlnorm` distribution we used to simulate the individual level reporting delay above, but here we use the cumulative distribution function rather than the probability density function. This gives us the probability that a report is made on day 1 or earlier, day 2 or earlier, etc.
+
+We can now construct some simulated data and use this delay distribution to nowcast the most recent data. Here we use the same simulation approach as in the [renewal session](R-estimation-and-the-renewal-equation) and apply the `reporting_delay` to the last 15 days of data.
+
+```{r, load-simulated-onset}
+source(here::here("snippets", "simulate-onsets.r"))
+reported_onset_df <- onset_df |>
+  filter(day < 40) |>
+  mutate(proportion_reported = c(rep(1, n() - 15), rev(proportion_reported)),
+         reported_onsets = map_dbl(onsets * proportion_reported, \(x) rpois(1, x)) 
+  )
```

Comment:
here I think it would be good to do this inside an R function `condition_onsets_by_report()` that does exactly the same as the stan function below.

Reply:
Part of #122?